src/openrct2/world/Park.cpp

```diff
@@ -768,7 +768,7 @@ void update_park_fences_around_tile(sint32 x, sint32 y)
 void park_set_name(const char *name)
 {
     // Required else the pointer arithmetic in the game commands below could cause an access violation
-    char* newName = (char *)malloc(USER_STRING_MAX_LENGTH + 1);
+    char* newName = (char *)malloc(USER_STRING_MAX_LENGTH + 5);
```

Comment:
Either `USER_STRING_MAX_LENGTH` is too small or the buffer allocated here, as just a few lines below we try writing 4 bytes at offset 32.